webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/Score.scala

```diff
@@ -0,0 +1,251 @@
+    }
+  }
+}
+
+object InventoryEventHandler extends ScoreEventHandler {
+  def handle(event: ScoreEvent): IOResult[List[(NodeId, List[Score[_]])]] = {
+
+    event match {
+      case InventoryScoreEvent(n, inventory) =>
+        val sum         = inventory.node.softwareUpdates.size
+        val security    = inventory.node.softwareUpdates.filter(_.kind == SoftwareUpdateKind.Security).size
+        val patch       = inventory.node.softwareUpdates.filter(_.kind == SoftwareUpdateKind.None).size
+        val defect      = inventory.node.softwareUpdates.filter(_.kind == SoftwareUpdateKind.Defect).size
+        val enhancement = inventory.node.softwareUpdates.filter(_.kind == SoftwareUpdateKind.Enhancement).size
+        val other       = inventory.node.softwareUpdates.filter { s =>
+          s.kind match {
+            case SoftwareUpdateKind.Other(_) => true
+            case _                           => false
+          }
+        }.size
+        val stats       = SystemUpdateStats(
+          sum,
+          if (security > 0) Some(security) else None,
+          if (patch > 0) Some(patch) else None,
+          if (defect > 0) Some(defect) else None,
+          if (enhancement > 0) Some(enhancement) else None,
+          if (other > 0) Some(other) else None
+        )
+        val score       = if (security == 0 && sum < 50) {
+          SystemUpdateScore(A, "Node has no security updates and less than 50 updates available", stats)
+        } else if (security < 5) {
+          SystemUpdateScore(B, s"Node has ${security} security updates available (less than 5)", stats)
+        } else if (security < 20) {
+          SystemUpdateScore(C, s"Node has ${security} security updates available (less than 20)", stats)
+        } else if (security < 50) {
+          SystemUpdateScore(D, s"Node has ${security} security updates available (less than 50)", stats)
+        } else {
+          SystemUpdateScore(E, s"Node has ${security} security updates available (more than 50)", stats)
+        }
+        ((n, score :: Nil) :: Nil).succeed
+      case _                                 => Nil.succeed
+    }
+  }
+}
+
+class ScoreService(globalScoreRepository: GlobalScoreRepository, scoreRepository: ScoreRepository) {
+  private[this] val cache:      Ref[Map[NodeId, GlobalScore]]    = globalScoreRepository.getAll().flatMap(Ref.make(_)).runNow
+  private[this] val scoreCache: Ref[Map[NodeId, List[Score[_]]]] = scoreRepository.getAll().flatMap(Ref.make(_)).runNow
+
+  def getAll():                       IOResult[Map[NodeId, GlobalScore]] = cache.get
+  def getGlobalScore(nodeId: NodeId): IOResult[GlobalScore]              = {
+    for {
+      c   <- cache.get
+      res <-
+        c.get(nodeId) match {
+          case Some(g) => g.succeed
+          case None    => Inconsistency("No global score for node").fail
+        }
+    } yield {
+      res
+    }
+  }
+
+  def getScoreDetails(nodeId: NodeId): IOResult[List[Score[_]]] = {
+    for {
```

Comment:
I think we should always return the list of score for ALL registered modules, and if the node is missing some score, have a default policy (perhaps configurable by score module). 
Else, if the node doesn't have a score (for ex because it was not computed yet), then it appears as not have it all all (and not like a missing score) ; plus the final score value won't be correct (we should have a way to tell "the value of that score is not computable yet because some data are missing", and not "C because it's the only data I have")

Reply:
discussed irl: 
- we need - in the future - way to decide for each node what score are enabled for it and only display these ones (and use these one for computation). Plus, some module could be optionnal, or only enabled on a given subset of nodes. 
- at some point in the future, we will have a "computation rule for that score" parameter, and it will perhaps need to be a per-node param, including the previous point ("enable/disable XXX score for that node")
- we still need a way to display a "not computed" score, ie a greyed/disabled version of score, which is a better feedback for the user than "nothing"
- we don't want for now to include disabled module as part of the computation

For now, only the last two points are relevant, but they are not blockers for that PR.